### ISSUE TITLE
Documentation of Time Formats

### ISSUE THREAD

**Issue by MaxNoe** [astropy/astropy#9925]
### Description

The documentation of the time scales and formats is lacking crucial information on how they are defined.

E.g. MJD is or JD just says: 
 
> This represents the number of days since midnight on November 17, 1858.
For example, 51544.0 in MJD is midnight on January 1, 2000.

Missing information is e.g.:
How are days defined here: calendar days? 86400 SI Seconds? 
Is it different for the different time scales? How do e.g. UTC leap seconds affect the fractional part of the MJD?

**Comment by bsipocz** [comment]
My limited understanding is that MJD and JD are formats, and the answer to your question is about the scale: http://docs.astropy.org/en/latest/time/#time-scale
A Time object has both of these.

This/similar certainly came up before, so if you have any suggestion how to make it clearer in the docs, please let us know, either here or preferably in a PR.

**Comment by MaxNoe** [comment]
No, I do know how leap seconds and the length of a second is handled in the scale.

I want to know, how that translates into the representation in a certain format.

So for UTC I found out, that the MJD  integer is counting calendar days, and the fractional part is the fraction of that day in SI Seconds by looking at 12:00 on a day with and without leap second:

```
In [1]: from astropy.time import Time

In [2]: Time('2016-12-31T12:00', scale='utc').mjd
Out[2]: 57753.49999421303

In [3]: Time('2016-12-30T12:00', scale='utc').mjd
Out[3]: 57752.5

```

TAI has no leap seconds and thus:
```
In [4]: Time('2016-12-30T12:00', scale='tai').mjd
Out[4]: 57752.5

In [5]: Time('2016-12-31T12:00', scale='tai').mjd
Out[5]: 57753.5
```

What is missing is an explanation, in what days are, if the reference date is changed to the corresponding scale (it seems so) and how the fractional part is calculated. (SI seconds of that day in that scale, what about UT1 that does not tick SI seconds)?

**Comment by pllim** [comment]
If only we can refactor how humanity measure time... And also get rid of the whole magnitude system.

**Comment by MaxNoe** [comment]
Yeah.

**Comment by mhvk** [comment]
On the other hands, days, hours, degrees, and magnitudes are the oldest units still in use - am somewhat happy the french revolution didn't quite succeed in replacing them!

More to the actual point: we basically follows IAU-SOFA, in which the day is not constant (in UTC), and hence one indeed gets fractions that do not mean the same thing. Bottom line is that it better not to use jd or mjd (or really anything but `iso`) with UTC. Suggestions on how to clarify this most welcome!!

**Comment by LoveChauhan-18** [comment]
Hello, I would like to contribute to this issue. Would it be possible to assign it to me?
